### PR TITLE
libraries: exclude the full set of libc6

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

**Comment by kyrofa** [comment]
I like this, but it seems like a bit of a half measure. In an ideal scenario, wouldn't we want to do this for every package included in the core snap? A list that, of course, we do not have.

File: snapcraft/internal/libraries.py [hunk 1]
```diff
@@ -73,9 +73,10 @@ def _get_system_libs():
         lib_path = None
 
     if not lib_path or not os.path.exists(lib_path):
-        logger.debug('No libraries to exclude from this release')
-        # Always exclude libc.so.6
-        return frozenset(['libc.so.6'])
+        logger.debug('Only excluding libc libraries from the release')
+        libc6_libs = [os.path.basename(l)
+                      for l in repo.Repo.get_package_libraries('libc6')]
```
Comment (kyrofa) [inline snapcraft/internal/libraries.py]: `get_package_libraries` definitely includes more than just libs. Will that cause issues, here?

**Comment by sergiusens** [comment]
On mié, oct 18, 2017 at 5:51 PM, Kyle Fazzari 
<notifications@github.com> wrote:
> I like this, but it seems like a bit of a half measure. In an ideal 
> scenario, wouldn't we want to do this for every package included in 
> the core snap? A list that, of course, we do not have.
> 

This is one step in the correct direction, we never want half of libc6 
consumed into the snap. The correct measure is to disable this 
functionality completely and never bring libraries in from the host.
>

**Comment by sergiusens** [comment]
On mié, oct 18, 2017 at 5:54 PM, Kyle Fazzari 
<notifications@github.com> wrote:
> @kyrofa commented on this pull request.
> 
> In snapcraft/internal/libraries.py:
> 
> > @@ -73,9 +73,10 @@ def _get_system_libs():
>          lib_path = None
> 
>      if not lib_path or not os.path.exists(lib_path):
> -        logger.debug('No libraries to exclude from this release')
> -        # Always exclude libc.so.6
> -        return frozenset(['libc.so.6'])
> +        logger.debug('Only excluding libc libraries from the 
> release')
> +        libc6_libs = [os.path.basename(l)
> +                      for l in 
> repo.Repo.get_package_libraries('libc6')]
> get_package_libraries definitely includes more than just libs. Will 
> that cause issues, here?
> 
It should really not, but what did it bring? You use this logic in dump 
and copy already for similar reasons I believe.
>

**Comment by kalikiana** [comment]
For the record, hitting the timeout again here `test_rust_plugin.RustPluginTestCase.test_cross_compiling ... 
The job exceeded the maximum time limit for jobs, and has been terminated.`

**Comment by kyrofa** [comment]
Ah ha, once I removed the `LD_LIBRARY_PATH` from `bin/snapcraft-classic` I was actually able to hit this error, and this PR fixes it.

**Comment by kalikiana** [comment]
@kyrofa What do you mean when you say "this error" here? Would it be worth linking a bug in a comment here? Or just a comment pointing out the problem?

**Comment by sergiusens** [comment]
#1635 was the other problem which caused initial suspicions on this PR.

**Pull request merged**